src/Microsoft.ApiDesignGuidelines.Analyzers/Core/IdentifiersShouldNotContainUnderscores.Fixer.cs

```diff
@@ -10,9 +10,10 @@ namespace Microsoft.ApiDesignGuidelines.Analyzers
     /// <summary>
     /// CA1707: Identifiers should not contain underscores
     /// </summary>
-    public abstract class IdentifiersShouldNotContainUnderscoresFixer : CodeFixProvider
+    public abstract class IdentifiersShouldNotContainUnderscoresFixer<TLanguageSyntaxKind> : CodeFixProvider
+        where TLanguageSyntaxKind : struct
     {
-        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(IdentifiersShouldNotContainUnderscoresAnalyzer.RuleId);
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(IdentifiersShouldNotContainUnderscoresAnalyzer<TLanguageSyntaxKind>.RuleId);
 
         public sealed override FixAllProvider GetFixAllProvider()
```

Comment:
I am not implementing the fixer now but the fixer can invoke a rename with the underscores removed, which is the fix mentioned in the MSDN page as well